server/src/main/java/org/elasticsearch/common/bytes/PagedBytesReference.java

```diff
@@ -39,11 +39,8 @@
     private final int offset;
     private final int length;
 
-    public PagedBytesReference(ByteArray byteArray, int length) {
-        this(byteArray, 0, length);
-    }
-
-    private PagedBytesReference(ByteArray byteArray, int from, int length) {
+    PagedBytesReference(ByteArray byteArray, int from, int length) {
+        assert byteArray.hasArray() == false;
```

Comment:
Just to remind me when I inevitably trip over this in future...

```suggestion
        assert byteArray.hasArray() == false : "use BytesReference#fromByteArray";
```